chain-registry/src/relayer_config.rs

```diff
@@ -0,0 +1,249 @@
+use tokio;
+use tokio::time::timeout;
+
+// ----------------- RPC ------------------
+
+/// Data retrievable from RPC endpoints
+#[derive(Clone, Debug)]
+pub struct RpcMandatoryData {
+    pub rpc_address: String,
+    pub max_block_size: u64,
+    pub websocket: tendermint_rpc::Url,
+    // max_block_time should also be retrieved from the RPC
+    // however it looks like it is not in the genesis file anymore
+}
+
+/// Retrieves the mandatory data from the RPC endpoint
+async fn query_rpc(rpc: &str) -> Result<RpcMandatoryData, RegistryError> {
+    let websocket_addr = websocket_from_rpc(rpc)?;
+
+    let (client, driver) = timeout(
+        Duration::from_secs(5),
+        WebSocketClient::new(websocket_addr.clone()),
+    )
+    .await
+    .map_err(|e| RegistryError::websocket_time_out_error(websocket_addr.to_string(), e))?
+    .map_err(|e| RegistryError::websocket_connect_error(websocket_addr.to_string(), e))?;
+
+    let driver_handle = tokio::spawn(async move { driver.run().await });
+
+    let latest_consensus_params = match client.latest_consensus_params().await {
+        Ok(response) => response.consensus_params.block.max_bytes,
+        Err(e) => {
+            return Err(RegistryError::rpc_consensus_params_error(
+                rpc.to_string(),
+                e,
+            ))
+        }
+    };
+
+    client.close().unwrap();
+    let _ = driver_handle.await.unwrap();
+
+    Ok(RpcMandatoryData {
+        rpc_address: rpc.to_string(),
+        max_block_size: latest_consensus_params,
+        websocket: websocket_addr,
+    })
+}
+
+// ----------------- websocket ------------------
+
+/// Generates a websocket address from a rpc address.
+fn websocket_from_rpc(rpc_endpoint: &str) -> Result<tendermint_rpc::Url, RegistryError> {
+    let uri = rpc_endpoint
+        .parse::<Uri>()
+        .map_err(|e| RegistryError::uri_parse_error(rpc_endpoint.to_string(), e))?;
+
+    let builder = Uri::builder();
+    match builder
+        .scheme("wss")
+        .authority(uri.authority().unwrap().clone())
+        .path_and_query("/websocket")
+        .build()
+    {
+        Ok(uri) => Ok(tendermint_rpc::Url::from_str(uri.to_string().as_str()).unwrap()),
+        Err(e) => Err(RegistryError::unable_to_build_websocket_endpoint(
+            rpc_endpoint.to_string(),
+            e,
+        )),
+    }
+}
+
+// ----------------- GRPC ------------------
+
+/// Returns a tendermint url if the client can connect to the gRPC server
+async fn health_check_grpc(grpc_address: &str) -> Result<tendermint_rpc::Url, RegistryError> {
+    let uri = parse_or_build_grpc_endpoint(grpc_address)?;
+    let tendermint_url = uri.to_string().parse().unwrap();
+    QueryClient::connect(uri)
+        .await
+        .map_err(|_| RegistryError::unable_to_connect_with_grpc())?;
+    Ok(tendermint_url)
+}
+
+/// Select a healthy rpc/grpc address from a list of urls
+async fn select_healthy<'a, RES, FUNC, FUTURE>(
```

Comment:
In Rust, multi-characters type parameters are typically written as CamelCase rather than uppercase.

```suggestion
async fn select_healthy<'a, Res, Func, Fut>(
```